packages/transformers/react-refresh-babel/src/ReactRefreshBabelTransformer.js

```diff
@@ -0,0 +1,27 @@
+// @flow
+
+import {Transformer} from '@parcel/plugin';
+import invariant from 'assert';
+
+function shouldExclude(asset, options) {
+  return !asset.env.isBrowser() || !options.hot || !asset.isSource;
+}
+
+export default new Transformer({
+  async transform({asset, options}) {
+    if (!shouldExclude(asset, options)) {
```

Comment:
What if it’s not a react app? Do we need to detect if `react` is in package.json or something?

Reply:
Done (and also safe if the babel transform has some bug)